### ISSUE TITLE
Note contents not appearing in reading mode when note has a Meta Bind button in it

### ISSUE THREAD

**Issue by laoraryn** [mProjectsCode/obsidian-meta-bind-plugin#390]
### Please fill out these Check-boxes

- [X] I checked that the plugin is up to date
- [X] The issue persist with all other plugins and themes disabled

### This Issue Occurs on

- [ ] Windows
- [X] Linux
- [ ] macOS
- [X] Android
- [ ] iOS

### Plugin Version

1.1.3

### Describe the Issue

I'm seeing an inconsistent issue where notes with Meta Bind buttons are not displaying their content in reading mode. The text appears in edit mode, but disappears in reading mode. 

It happens in some (but not all) notes I have with buttons in them, and not in notes that don't have buttons. It sometimes spontaneously starts working, but I haven't noticed a pattern. I noticed it in my daily notes, but when I copy-pasted the text from a problem note into a brand new note, the issue persisted.

I created a screen recording to show the issue - I tried to attach it here, but if that doesn't work, I can share a Google Drive link:

https://github.com/user-attachments/assets/f9e8fe20-9a97-4326-8ae7-494ae352f4d3

### Steps to Reproduce

1. Create a note with a Meta Bind button in it
2. Switch from source mode to reading mode

I isolated the issue to the Meta Bind plugin, and disabled all other plugins, my theme, and css snippets. It still persisted. Weirdly, when I was turning on/off plugins, turning off the Excalidraw plugin could also cause the note to reappear, but it wasn't consistent. Possibly it's because it caused the active note to refresh?

I'm using Obsidian on Linux Mint 21.3 Cinnamon. It's also occurring on my phone (Google Pixel 8 Pro, Android 14), and I think it was also occurring when I was running Windows a couple weeks ago, but I can't remember for sure.

### Expected Behavior

I'd expect note contents to show consistently when switching to reading mode.

**Comment by mProjectsCode** [comment]
Can you check the console for any errors?

**Comment by laoraryn** [comment]
Forgot that was an option! May have found the problem - getting a 'button with id _ already exists in the button templates' error. I had been putting the button code at the bottom of each note, so I could do inline buttons above. When I found the template option in settings, I switched to that but never removed the code at the bottom of the notes.

When I remove the extra code, the notes show up consistently! So chalk this one up to user error :)

Full error from the console below:

`<div aria-expanded="true" tabindex="-1" style="box-sizing: border-box; min-width: 0px; min-height: 0px; outline-width: 0px; flex: 0 0 auto; color: rgb(227, 227, 227); font-family: &quot;Noto Sans Mono&quot;, &quot;DejaVu Sans Mono&quot;, monospace; font-size: 11px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(78, 53, 52); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span class="source-code" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--source-code-font-family); white-space: pre-wrap; font-size: var(--source-code-font-size) !important; line-height: 1.2;"><span class="console-message-anchor" style="box-sizing: border-box; min-width: 0px; min-height: 0px; float: right; text-align: right; max-width: 100%; margin-left: 4px;"><button class="devtools-link text-button link-style" role="link" tabindex="-1" title="app://obsidian.md/plugin:obsidian-meta-bind-plugin:57" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font: inherit; margin: 0px; height: unset; border: none; border-radius: 2px; padding: 0px !important; color: var(--sys-color-primary); background: none; flex: 0 0 auto; white-space: nowrap; text-decoration: underline; outline-offset: 2px; outline: none; cursor: pointer; word-break: break-all;"><br class="Apple-interchange-newline">plugin:obsidian-meta-bind-plugin:57</button> </span><span class="console-message-text" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--override-error-text-color) !important;">Uncaught <span class="object-value-error source-code" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--source-code-font-family); white-space: pre-wrap; font-size: var(--source-code-font-size) !important; line-height: 1.2;"><span style="box-sizing: border-box; min-width: 0px; min-height: 0px;">Error: ButtonManager | button with id "thing" already exists in the button templates
<span style="box-sizing: border-box; min-width: 0px; min-height: 0px;">    at Bp.addButton (<button class="devtools-link text-button link-style" role="link" tabindex="-1" title="app://obsidian.md/plugin:obsidian-meta-bind-plugin:57:18728" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font: inherit; margin: 0px; height: unset; border: none; border-radius: 2px; padding: 0px !important; color: var(--sys-color-primary); background: none; flex: 0 0 auto; white-space: nowrap; text-decoration: underline; outline-offset: 2px; outline: none; cursor: pointer; word-break: break-all;">plugin:obsidian-meta<span class="devtools-link-ellipsis" style="box-sizing: border-box; min-width: 0px; min-height: 0px;">…</span>ind-plugin:57:18728</button>)
</span><span style="box-sizing: border-box; min-width: 0px; min-height: 0px;">    at es.onMount (<button class="devtools-link text-button link-style" role="link" tabindex="-1" title="app://obsidian.md/plugin:obsidian-meta-bind-plugin:57:14248" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font: inherit; margin: 0px; height: unset; border: none; border-radius: 2px; padding: 0px !important; color: var(--sys-color-primary); background: none; flex: 0 0 auto; white-space: nowrap; text-decoration: underline; outline-offset: 2px; outline: none; cursor: pointer; word-break: break-all;">plugin:obsidian-meta<span class="devtools-link-ellipsis" style="box-sizing: border-box; min-width: 0px; min-height: 0px;">…</span>ind-plugin:57:14248</button>)
</span><span style="box-sizing: border-box; min-width: 0px; min-height: 0px;">    at es.mount (<button class="devtools-link text-button link-style" role="link" tabindex="-1" title="app://obsidian.md/plugin:obsidian-meta-bind-plugin:57:606" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font: inherit; margin: 0px; height: unset; border: none; border-radius: 2px; padding: 0px !important; color: var(--sys-color-primary); background: none; flex: 0 0 auto; white-space: nowrap; text-decoration: underline; outline-offset: 2px; outline: none; cursor: pointer; word-break: break-all;">plugin:obsidian-meta-bind-plugin:57:606</button>)
</span><span style="box-sizing: border-box; min-width: 0px; min-height: 0px;">    at Pp.onMount (<button class="devtools-link text-button link-style" role="link" tabindex="-1" title="app://obsidian.md/plugin:obsidian-meta-bind-plugin:57:16773" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font: inherit; margin: 0px; height: unset; border: none; border-radius: 2px; padding: 0px !important; color: var(--sys-color-primary); background: none; flex: 0 0 auto; white-space: nowrap; text-decoration: underline; outline-offset: 2px; outline: none; cursor: pointer; word-break: break-all;">plugin:obsidian-meta<span class="devtools-link-ellipsis" style="box-sizing: border-box; min-width: 0px; min-height: 0px;">…</span>ind-plugin:57:16773</button>)
</span><span style="box-sizing: border-box; min-width: 0px; min-height: 0px;">    at Pp.mount (<button class="devtools-link text-button link-style" role="link" tabindex="-1" title="app://obsidian.md/plugin:obsidian-meta-bind-plugin:57:606" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font: inherit; margin: 0px; height: unset; border: none; border-radius: 2px; padding: 0px !important; color: var(--sys-color-primary); background: none; flex: 0 0 auto; white-space: nowrap; text-decoration: underline; outline-offset: 2px; outline: none; cursor: pointer; word-break: break-all;">plugin:obsidian-meta-bind-plugin:57:606</button>)
</span><span style="box-sizing: border-box; min-width: 0px; min-height: 0px;">    at Lh.onload (<button class="devtools-link text-button link-style" role="link" tabindex="-1" title="app://obsidian.md/plugin:obsidian-meta-bind-plugin:111:19428" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font: inherit; margin: 0px; height: unset; border: none; border-radius: 2px; padding: 0px !important; color: var(--sys-color-primary); background: none; flex: 0 0 auto; white-space: nowrap; text-decoration: underline; outline-offset: 2px; outline: none; cursor: pointer; word-break: break-all;">plugin:obsidian-meta<span class="devtools-link-ellipsis" style="box-sizing: border-box; min-width: 0px; min-height: 0px;">…</span>nd-plugin:111:19428</button>)
</span><span style="box-sizing: border-box; min-width: 0px; min-height: 0px;">    at e.load (<button class="devtools-link text-button link-style" role="link" tabindex="-1" title="app://obsidian.md/app.js:1:1166456" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font: inherit; margin: 0px; height: unset; border: none; border-radius: 2px; padding: 0px !important; color: var(--sys-color-primary); background: none; flex: 0 0 auto; white-space: nowrap; text-decoration: underline; outline-offset: 2px; outline: none; cursor: pointer; word-break: break-all;">app.js:1:1166456</button>)
</span><span style="box-sizing: border-box; min-width: 0px; min-height: 0px;">    at e.addChild (<button class="devtools-link text-button link-style" role="link" tabindex="-1" title="app://obsidian.md/app.js:1:1166857" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font: inherit; margin: 0px; height: unset; border: none; border-radius: 2px; padding: 0px !important; color: var(--sys-color-primary); background: none; flex: 0 0 auto; white-space: nowrap; text-decoration: underline; outline-offset: 2px; outline: none; cursor: pointer; word-break: break-all;">app.js:1:1166857</button>)
</span><span style="box-sizing: border-box; min-width: 0px; min-height: 0px;">    at Object.addChild (<button class="devtools-link text-button link-style" role="link" tabindex="-1" title="app://obsidian.md/app.js:1:1296415" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font: inherit; margin: 0px; height: unset; border: none; border-radius: 2px; padding: 0px !important; color: var(--sys-color-primary); background: none; flex: 0 0 auto; white-space: nowrap; text-decoration: underline; outline-offset: 2px; outline: none; cursor: pointer; word-break: break-all;">app.js:1:1296415</button>)
</span><span style="box-sizing: border-box; min-width: 0px; min-height: 0px;">    at Vh.wrapInMDRC (<button class="devtools-link text-button link-style" role="link" tabindex="-1" title="app://obsidian.md/plugin:obsidian-meta-bind-plugin:111:20078" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font: inherit; margin: 0px; height: unset; border: none; border-radius: 2px; padding: 0px !important; color: var(--sys-color-primary); background: none; flex: 0 0 auto; white-space: nowrap; text-decoration: underline; outline-offset: 2px; outline: none; cursor: pointer; word-break: break-all;">plugin:obsidian-meta<span class="devtools-link-ellipsis" style="box-sizing: border-box; min-width: 0px; min-height: 0px;">…</span>nd-plugin:111:20078</button>)</span></span></span></span></span></div><div class="" role="group" style="box-sizing: border-box; min-width: 0px; min-height: 0px; flex: 0 0 auto; color: rgb(227, 227, 227); font-family: &quot;Noto Sans Mono&quot;, &quot;DejaVu Sans Mono&quot;, monospace; font-size: 11px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(78, 53, 52); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span class="monospace stack-preview-container width-constrained" style="display: inline-block; --monospace-font-size: 11px; --monospace-font-family: &quot;Noto Sans Mono&quot;,&quot;DejaVu Sans Mono&quot;,monospace; --source-code-font-size: 11px; --source-code-font-family: &quot;Noto Sans Mono&quot;,&quot;DejaVu Sans Mono&quot;,monospace; width: 485px; box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--monospace-font-family); font-size: var(--monospace-font-size) !important;">
  | addButton | @ | plugin:obsidian-meta-bind-plugin:57
-- | -- | -- | --
  | onMount | @ | plugin:obsidian-meta-bind-plugin:57
  | mount | @ | plugin:obsidian-meta-bind-plugin:57
  | onMount | @ | plugin:obsidian-meta-bind-plugin:57
  | mount | @ | plugin:obsidian-meta-bind-plugin:57
  | onload | @ | plugin:obsidian-meta-bind-plugin:111
  | e.load | @ | app.js:1
  | e.addChild | @ | app.js:1
  | addChild | @ | app.js:1
  | wrapInMDRC | @ | plugin:obsidian-meta-bind-plugin:111
  | eval | @ | plugin:obsidian-meta-bind-plugin:133
  | (anonymous) | @ | app.js:1
  | t.postProcess | @ | app.js:1
  | t.postProcess | @ | app.js:1
  | h | @ | app.js:1
  | e.onRender | @ | app.js:1
  | requestAnimationFrame (async) |   |  
  | iu | @ | app.js:1
  | e.queueRender | @ | app.js:1
  | (anonymous) | @ | app.js:1
  | r | @ | enhance.js:1

</span></div>`
